pkg/cri/instrument/instrumented_service.go

```diff
@@ -19,21 +19,28 @@ package instrument
 import (
 	"context"
 	"errors"
+	"sync"
 
 	"github.com/containerd/containerd/errdefs"
 	"github.com/containerd/containerd/log"
+	"github.com/containerd/containerd/services/warning"
 	runtime_alpha "github.com/containerd/containerd/third_party/k8s.io/cri-api/pkg/apis/runtime/v1alpha2"
 	"github.com/containerd/containerd/tracing"
 	runtime "k8s.io/cri-api/pkg/apis/runtime/v1"
 
 	ctrdutil "github.com/containerd/containerd/pkg/cri/util"
+	"github.com/containerd/containerd/pkg/deprecation"
 )
 
 const (
 	// criSpanPrefix is a prefix for CRI server specific spans
 	criSpanPrefix = "pkg.cri.server"
 )
 
+var (
+	emitWarning sync.Once
+)
```

Comment:
Let's move this to be a field in `instrumentedAlphaService` rather than a package-level variable.

Reply:
Done